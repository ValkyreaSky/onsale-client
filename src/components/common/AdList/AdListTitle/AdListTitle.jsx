import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.h2`
  margin: 2rem 0;
  font-size: 2.2rem;
`;

const SmallTitle = styled(Title)`
  font-size: 1.7rem;
`;

const AdListTitle = ({ text, small }) => {
  if (small) {
    return <SmallTitle>{text}</SmallTitle>;
  }
  return <Title>{text}</Title>;
};

AdListTitle.defaultProps = {
  small: false,
};

AdListTitle.propTypes = {
  text: PropTypes.string.isRequired,
  small: PropTypes.bool,
};

export { Title, SmallTitle };
export default AdListTitle;
