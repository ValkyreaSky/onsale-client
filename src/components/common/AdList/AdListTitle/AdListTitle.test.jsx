import React from 'react';
import { shallow } from 'enzyme';
import AdListTitle, { Title, SmallTitle } from './AdListTitle';

describe('<AdListTitle />', () => {
  const text = 'Hello';

  it('should match snapthot', () => {
    const wrapper = shallow(<AdListTitle text={text} />);
    expect(wrapper).toMatchSnapshot();
  });


  it('should render <Title> component when "small" prop is false', () => {
    const wrapper = shallow(<AdListTitle text={text} />);
    expect(wrapper.find(Title).length).toBe(1);
    expect(wrapper.find(SmallTitle).length).toBe(0);
    expect(wrapper.find(Title).props().children).toBe(text);
  });

  it('should render <SmallTitle> when "small" prop is true', () => {
    const wrapper = shallow(<AdListTitle small text={text} />);
    expect(wrapper.find(SmallTitle).length).toBe(1);
    expect(wrapper.find(Title).length).toBe(0);
    expect(wrapper.find(SmallTitle).props().children).toBe(text);
  });
});
