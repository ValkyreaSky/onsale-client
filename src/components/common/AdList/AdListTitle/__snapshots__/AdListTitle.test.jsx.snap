// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdListTitle /> should match snapthot 1`] = `
<styled.h2>
  Hello
</styled.h2>
`;
